dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/memory/RelCo.java

```diff
@@ -220,7 +217,47 @@ protected BooleanFormula encodeApprox(SolverContext ctx) {
         }
         return enc;
     }
-    
+
+    /*
+        Returns a set of co-edges (w1, w2) (subset of maxTupleSet) whose clock-constraints
+        do not need to get encoded explicitly.
+        The reason is that whenever we have co(w1,w2) then there exists an intermediary
```

Comment:
"there exists" should be "there might exist", right? The `co` edge might actually be a direct one.

Reply:
No, the explanation is exactly for those co-edges that do not need to get encoded (and for those there exists such an intermediary). I can reformulate the sentence a little bit to make this more clear.

Reply:
Now I get it. The `co(W1,W2)` in the second sentence refers to those defined in the first sentence and not any pair.